Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/HeadsetRequestHandler.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.Utilities;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    public class HeadsetRequestHandler : MonoBehaviour
+    {
+        /// <summary>
+        /// Used to obtain marker, headset and pv camera data.
+        /// </summary>
+        [Tooltip("Used to obtain marker, headset and pv camera data.")]
+        [SerializeField]
+        private HeadsetCalibration headsetCalibration;
+
+        /// <summary>
+        /// Used to setup a network connection.
+        /// </summary>
+        [Tooltip("Used to setup a network connection.")]
+        [SerializeField]
+        private MonoBehaviour MatchMakingService;
+        private IMatchMakingService matchMakingService;
+
+        /// <summary>
+        /// Used to send/receive data related to the calibration process.
+        /// </summary>
+        [Tooltip("Used to send/receive data related to the calibration process.")]
+        [SerializeField]
+        private MonoBehaviour NetworkingService;
+        private INetworkingService networkingService;
+
+        /// <summary>
+        /// Used to display the last request timestamp.
+        /// </summary>
+        [Tooltip("Used to display the last request timestamp.")]
+        [SerializeField]
+        private Text lastRequestTimestampText;
+
+        private bool updateData = false;
+        private HeadsetCalibrationDataRequest request = null;
+
+        private void OnValidate()
+        {
+#if UNITY_EDITOR
```

Comment:
Thoughts on moving this around OnValidate() so that it's purely an editor thing?